Error at linking with cmake: undefined reference to `disp_spi_send_xxx`

I followed the installation instructions from the readme and installed the latest version of lv_port_esp32 as submodule for my ESP-IDF Projekt.
When I compile with cmake (idf.py build) I got following error at linking:

```
/home/tobiaskuch/.espressif/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.2.0/../../../../xtensa-esp32-elf/bin/ld: esp-idf/lvgl_tft/liblvgl_tft.a(ili9341.c.obj):(.literal.ili9341_send_data+0x4): undefined reference to `disp_spi_send_data'
/home/tobiaskuch/.espressif/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.2.0/../../../../xtensa-esp32-elf/bin/ld: esp-idf/lvgl_tft/liblvgl_tft.a(ili9341.c.obj):(.literal.ili9341_flush+0x0): undefined reference to `disp_spi_send_colors'
/home/tobiaskuch/.espressif/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.2.0/../../../../xtensa-esp32-elf/bin/ld: esp-idf/lvgl_tft/liblvgl_tft.a(ili9341.c.obj): in function `ili9341_send_data':
/home/tobiaskuch/esp/projects/shop_device/components/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/ili9341.c:206: undefined reference to `disp_spi_send_data'
/home/tobiaskuch/.espressif/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.2.0/../../../../xtensa-esp32-elf/bin/ld: esp-idf/lvgl_tft/liblvgl_tft.a(ili9341.c.obj): in function `ili9341_send_cmd':
/home/tobiaskuch/esp/projects/shop_device/components/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/ili9341.c:199: undefined reference to `disp_spi_send_data'
/home/tobiaskuch/.espressif/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.2.0/../../../../xtensa-esp32-elf/bin/ld: esp-idf/lvgl_tft/liblvgl_tft.a(ili9341.c.obj): in function `ili9341_flush':
/home/tobiaskuch/esp/projects/shop_device/components/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/ili9341.c:157: undefined reference to `disp_spi_send_colors'
collect2: error: ld returned 1 exit status
CMakeFiles/shop_device.elf.dir/build.make:368: recipe for target 'shop_device.elf' failed
make[2]: *** [shop_device.elf] Error 1
CMakeFiles/Makefile2:604: recipe for target 'CMakeFiles/shop_device.elf.dir/all' failed
make[1]: *** [CMakeFiles/shop_device.elf.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
make failed with exit code 2
```

I also changed the lv_port_esp32 cmake file like in the readme told.

```
cmake_minimum_required(VERSION 3.5)

#include($ENV{IDF_PATH}/tools/cmake/project.cmake)
list(APPEND EXTRA_COMPONENT_DIRS components/lvgl_esp32_drivers components/lvgl_esp32_drivers/lvgl_touch components/lvgl_esp32_drivers/lvgl_tft)

if (NOT DEFINED PROJECT_NAME)
	project(lvgl-demo)
endif (NOT DEFINED PROJECT_NAME)
```

This is my project root cmake.txt file

```
cmake_minimum_required(VERSION 3.5)

include($ENV{IDF_PATH}/tools/cmake/project.cmake)
set(EXTRA_COMPONENT_DIRS components/lv_port_esp32/components/lv_examples components/lv_port_esp32/components/lvgl components/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft components/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_touch components/lv_port_esp32/components/lvgl_esp32_drivers)
project(shop_device)
```

**ESP-IDF version**
release/v4.0

**Development kit used**
ESP_WROVER-KIT v4.1

**Development machine OS**
ubuntu18.04

**Complete Build log**
[build-log.log](https://github.com/lvgl/lv_port_esp32/files/4868418/build-log.log)

Hi @nikola-winmaker and @tobiwan42,

I was able to reproduce the issue when setting the optimization to -Os, but the error I got is, in the ST7789.c file:

```
/home/user/.espressif/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.2.0/../../../../xtensa-esp32-elf/bin/ld: esp-idf/lvgl_tft/liblvgl_tft.a(st7789.c.obj):(.literal.st7789_send_cmd+0x0): undefined reference to `disp_spi_send_data'
/home/user/.espressif/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.2.0/../../../../xtensa-esp32-elf/bin/ld: esp-idf/lvgl_tft/liblvgl_tft.a(st7789.c.obj):(.literal.st7789_flush+0x0): undefined reference to `disp_spi_send_colors'
/home/user/.espressif/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.2.0/../../../../xtensa-esp32-elf/bin/ld: esp-idf/lvgl_tft/liblvgl_tft.a(st7789.c.obj): in function `st7789_send_cmd':
/home/user/lv_port_esp32/build/../components/lvgl_esp32_drivers/lvgl_tft/st7789.c:192: undefined reference to `disp_spi_send_data'
/home/user/.espressif/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.2.0/../../../../xtensa-esp32-elf/bin/ld: esp-idf/lvgl_tft/liblvgl_tft.a(st7789.c.obj): in function `st7789_send_data':
/home/user/lv_port_esp32/build/../components/lvgl_esp32_drivers/lvgl_tft/st7789.c:199: undefined reference to `disp_spi_send_data'
/home/user/.espressif/tools/xtensa-esp32-elf/esp-2020r1-8.2.0/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/8.2.0/../../../../xtensa-esp32-elf/bin/ld: esp-idf/lvgl_tft/liblvgl_tft.a(st7789.c.obj): in function `st7789_send_color':
/home/user/lv_port_esp32/build/../components/lvgl_esp32_drivers/lvgl_tft/st7789.c:204: undefined reference to `disp_spi_send_colors'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
ninja failed with exit code 1
```

After some googling I found this stack overflow answer: https://stackoverflow.com/a/9399539

> The best place to put inline function is in a header file, and declare them static inline. This removes any need for an external definition, so it resolves the linker problem. However, this causes the compiler to emit the code for the function in every compilation unit that uses it, so could result in code bloat. But since the function is inline, it is probably small anyway, so this usually isn't a problem.

So in `disp_spi.h` I changed the following (added static to both inline functions):
```diff
diff --git a/components/lvgl_esp32_drivers/lvgl_tft/disp_spi.h b/components/lvgl_esp32_drivers/lvgl_tft/disp_spi.h
index 1a158a9..64da1c4 100644
--- a/components/lvgl_esp32_drivers/lvgl_tft/disp_spi.h
+++ b/components/lvgl_esp32_drivers/lvgl_tft/disp_spi.h
@@ -64,11 +64,11 @@ void disp_wait_for_pending_transactions(void);
 void disp_spi_acquire(void);
 void disp_spi_release(void);

-inline void disp_spi_send_data(uint8_t *data, size_t length) {
+static inline void disp_spi_send_data(uint8_t *data, size_t length) {
     disp_spi_transaction(data, length, DISP_SPI_SEND_POLLING, NULL, 0);
 }

-inline void disp_spi_send_colors(uint8_t *data, size_t length) {
+static inline void disp_spi_send_colors(uint8_t *data, size_t length) {
     disp_spi_transaction(data, length,
         DISP_SPI_SEND_QUEUED | DISP_SPI_SIGNAL_FLUSH,
         NULL, 0);
```

Do you think this is the most appropiate fix @nikola-winmaker ?

--- a/components/lvgl_esp32_drivers/lvgl_tft/disp_spi.h
+++ b/components/lvgl_esp32_drivers/lvgl_tft/disp_spi.h
@@ -64,11 +64,11 @@ void disp_wait_for_pending_transactions(void);
 void disp_spi_acquire(void);
 void disp_spi_release(void);
 
-inline void disp_spi_send_data(uint8_t *data, size_t length) {
+static inline void disp_spi_send_data(uint8_t *data, size_t length) {
     disp_spi_transaction(data, length, DISP_SPI_SEND_POLLING, NULL, 0);
 }
 
-inline void disp_spi_send_colors(uint8_t *data, size_t length) {
+static inline void disp_spi_send_colors(uint8_t *data, size_t length) {
     disp_spi_transaction(data, length,
         DISP_SPI_SEND_QUEUED | DISP_SPI_SIGNAL_FLUSH,
         NULL, 0);

